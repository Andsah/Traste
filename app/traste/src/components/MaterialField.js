/* eslint-disable react/prop-types */
import React from "react";

import { TextField, InputAdornment } from "@mui/material";

function MaterialField({ label, value, onChange, error }) {
  return (
    <TextField
      label={label}
      InputProps={{
        endAdornment: (
          <InputAdornment position="start" disableTypography={true}>
            %
          </InputAdornment>
        ),
      }}
      variant="outlined"
      type="number"
      onFocus={(event) => {
        event.target.select();
      }}
      onBlur={(evt) => {
        if (evt.target.value === "" || evt.target.value < 0) {
          onChange(0);
        }
      }}
      value={value}
      onChange={(e) => {
        var tmpval = e.target.value;
        if (isNaN(parseInt(e.target.value, 10))) {
          tmpval = "";
        } else {
          tmpval = parseInt(tmpval, 10);
        }
        onChange(tmpval);
      }}
      sx={{
        backgroundColor: "rgba(255,255,255,0.3)",
      }}
      inputProps={{
        style: { textAlign: "right", fontSize: 16 },
        "data-testid": "materialfield",
      }}
      error={!!error}
      //helperText={error ? error.message : null}
    />
  );
}
export default MaterialField;
