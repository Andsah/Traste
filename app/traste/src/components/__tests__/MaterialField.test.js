import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import MaterialField from '../MaterialField';
import React from 'react';

afterEach(() => {
  cleanup();
});

test('Should render MaterialField with name Wood', () => {
  const propsData = {name: 'Wood'};
  render(<MaterialField {...propsData} />);
  const materialFieldElement = screen.getByTestId('materialfield');
  expect(materialFieldElement).toBeInTheDocument();
  expect(materialFieldElement).toHaveTextContent('');
  expect(materialFieldElement).toHaveAttribute('name', 'Wood');
});

test('Should not render Material with name Wood', () => {
  const propsData = {label: 'Wood'};
  render(<MaterialField {...propsData} />);
  const materialFieldElement = screen.getByTestId('materialfield');
  expect(materialFieldElement).toBeInTheDocument();
  expect(materialFieldElement).not.toHaveAttribute('name', '');
});

test('Matches snapshot', () => {
  const tree = renderer.create(<MaterialField />).toJSON();
  expect(tree).toMatchSnapshot();
});
