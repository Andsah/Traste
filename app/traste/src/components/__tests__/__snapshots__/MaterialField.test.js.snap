// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1q0uy3-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid="materialfield"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      style={
        Object {
          "fontSize": 16,
          "textAlign": "right",
        }
      }
      type="number"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
      >
        ​
      </span>
      %
    </div>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-nnbavb"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
